### PR TITLE
feat: add first repo verticle

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #1099 
Added first verticle for repo configs and swagger-ui.

Swagger is available at `localhost:8081/api/index.html`. What difficulties we have:
1. org/flat modes, we need to figure out how we handle it because of extra username path parameter for `org` layout. Yaml open api schema does allow optional path parameters
2. now I have not figured out how to properly start artipie and api on the same port. In such case all the requests seem to be forwarded to artipie (not verticle) by vert.x. 

To check swagger and api you can start artipie locally, use `src/main/resources/example/artipie.yaml` config, but first, change storage path in it into your absolute path to `src/main/resources/example/repo` folder.

### CODE REVIEW

File: src/main/java/com/artipie/api/ManageRepoSettings.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     * @param key Key (ROOT or user name)
+     * @return List of the repositories
+     */
+    private Collection<String> list(final Key key) {
+        final Collection<String> res = new ArrayList<>(5);
+        for (final Key item : this.asto.list(key)) {
+            final String name = item.string();
+            // @checkstyle BooleanExpressionComplexityCheck (5 lines)
+            if ((name.endsWith(".yaml") || name.endsWith(".yml"))
+                && !name.contains("_storages")
```
Comment (dgarus) [inline src/main/java/com/artipie/api/ManageRepoSettings.java:87]: `_storages` looks like constant with according javadoc.

File: src/main/resources/swagger-ui/yaml/repository.yaml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: Swagger Petstore
+  license:
+    name: MIT
+paths:
+  /repository/list:
```
Comment (dgarus) [inline src/main/resources/swagger-ui/yaml/repository.yaml:8]: Should we add in paths version `v1/`?

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
Please have a look at my comments.

**Comment by olenagerasimova** [comment]
@dgarus thanks, corrected

**Review by dgarus — DISMISSED** [review]
@olenagerasimova thanks! LGTM

**Review by andpopov — COMMENTED** [review]
@olenagerasimova  Looks great!
Please see my little comments in code.

**Review by andpopov — DISMISSED** [review]

File: src/main/java/com/artipie/VertxMain.java [hunk 3]
```diff
@@ -106,6 +109,14 @@ public int start() throws IOException {
         );
         Logger.info(VertxMain.class, "Artipie was started on port %d", main);
         this.startRepos(settings, metrics, this.port);
+        this.vertx.deployVerticle(
+            new RepositoryVerticle(
+                new ManageRepoSettings(new BlockingStorage(settings.repoConfigsStorage())),
+                settings.layout().toString(),
+                // @checkstyle MagicNumberCheck (1 line)
+                8085
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/VertxMain.java]: it's hard-coded just for not, later I will figure out if it's possible to start artipie and verticles on the same port and extract app parameter.

**Comment by olenagerasimova** [comment]
@andpopov thanks, corrected and unswered

**Review by dgarus — APPROVED** [review]

**Review by andpopov — APPROVED** [review]

**Pull request merged**